Zend/zend_call_stack.c

```diff
@@ -450,12 +452,49 @@ static bool zend_call_stack_get_openbsd(zend_call_stack *stack)
 	return true;
 }
 #else
-static bool zend_call_stack_get_openbsd(zend_call_stack *stack)
+static bool zend_call_stack_get_openbsd_pthread(zend_call_stack *stack)
 {
 	return false;
 }
 #endif /* defined(HAVE_PTHREAD_STACKSEG_NP) */
 
+static bool zend_call_stack_get_openbsd_vm(zend_call_stack *stack)
+{
+	void *stack_base;
+	struct rlimit rlim;
+
+
+	if (getrlimit(RLIMIT_STACK, &rlim) != 0) {
+		return false;
+	}
+
+	if (rlim.rlim_cur == RLIM_INFINITY) {
+		return false;
+	}
+
+	// arch dependent top user's stack
+	stack->base = USRSTACK - rlim.rlim_cur;
+	stack->max_size = rlim.rlim_cur - sysconf(_SC_PAGE_SIZE);
+
+	return true;
+}
+
+static bool zend_call_stack_get_openbsd(zend_call_stack *stack)
+{
+	if (getthrid() == getpid()) {
```

Comment:
This trick doesn't appear to work on my machine: the IDs a different (I've verified that there is a single thread in the process). I'm not sure how we can determine that we are the main thread. From reading the code, `TIB_GET()->tib_thread == NULL || !TIB_GET()->tib_thread->stack` would work, but you may know a better way?